pandas/tests/arrays/floating/test_construction.py

```diff
@@ -0,0 +1,221 @@
+
+    result = pd.array([1, None, 2.0])
+    expected = pd.array([1.0, None, 2.0], dtype="Float64")
+    tm.assert_extension_array_equal(result, expected)
+
+
+@pytest.mark.parametrize(
+    "values",
+    [
+        ["foo", "bar"],
+        ["1", "2"],
+        "foo",
+        1,
+        1.0,
+        pd.date_range("20130101", periods=2),
+        np.array(["foo"]),
+        [[1, 2], [3, 4]],
+        [np.nan, {"a": 1}],
+    ],
+)
+def test_to_array_error(values):
+    # error in converting existing arrays to FloatingArray
+    msg = (
+        r"(:?.* cannot be converted to a FloatingDtype)"
+        r"|(:?values must be a 1D list-like)"
+        r"|(:?Cannot pass scalar)"
+    )
+    with pytest.raises((TypeError, ValueError), match=msg):
+        pd.array(values, dtype="Float64")
+
+
+def test_to_array_inferred_dtype():
+    # if values has dtype -> respect it
+    result = pd.array(np.array([1, 2], dtype="float32"))
+    assert result.dtype == Float32Dtype()
+
+    # if values have no dtype -> always float64
+    result = pd.array([1.0, 2.0])
+    assert result.dtype == Float64Dtype()
+
+
+def test_to_array_dtype_keyword():
+    result = pd.array([1, 2], dtype="Float32")
+    assert result.dtype == Float32Dtype()
+
+    # if values has dtype -> override it
+    result = pd.array(np.array([1, 2], dtype="float32"), dtype="Float64")
+    assert result.dtype == Float64Dtype()
+
+
+def test_to_array_integer():
+    result = pd.array([1, 2], dtype="Float64")
+    expected = pd.array([1.0, 2.0], dtype="Float64")
+    tm.assert_extension_array_equal(result, expected)
+
+    # for integer dtypes, the itemsize is not preserved
+    # TODO can we specify "floating" in general?
+    result = pd.array(np.array([1, 2], dtype="int32"), dtype="Float64")
+    assert result.dtype == Float64Dtype()
+
+
+@pytest.mark.parametrize(
+    "bool_values, values, target_dtype, expected_dtype",
+    [
+        ([False, True], [0, 1], Float64Dtype(), Float64Dtype()),
+        ([False, True], [0, 1], "Float64", Float64Dtype()),
+        ([False, True, np.nan], [0, 1, np.nan], Float64Dtype(), Float64Dtype()),
+    ],
+)
+def test_to_array_bool(bool_values, values, target_dtype, expected_dtype):
+    result = pd.array(bool_values, dtype=target_dtype)
+    assert result.dtype == expected_dtype
+    expected = pd.array(values, dtype=target_dtype)
+    tm.assert_extension_array_equal(result, expected)
+
+
+def test_series_from_float(data):
+    # construct from our dtype & string dtype
+    dtype = data.dtype
+
+    # from float
+    expected = pd.Series(data)
+    result = pd.Series(data.to_numpy(na_value=np.nan, dtype="float"), dtype=str(dtype))
+    tm.assert_series_equal(result, expected)
+
+    # from list
+    expected = pd.Series(data)
+    result = pd.Series(np.array(data).tolist(), dtype=str(dtype))
+    tm.assert_series_equal(result, expected)
+
+
+# TODO belongs in different file
+
+# def test_conversions(data_missing):
```

Comment:
can this (and anything in the same vein) be addressed before merging

Reply:
Ah, yes, this I certainly still need to clean up

Reply:
cleaned up those tests